### ISSUE TITLE
Issue validating Bootstrap 

### ISSUE THREAD

**Issue by mattphung** [sagrawal14/bootstrap-angular-validation#17]
I am having issues validating radio buttons using Bootstrap's button plugins below.
http://v4-alpha.getbootstrap.com/components/buttons/#checkbox-and-radio-buttons 

I'm trying to validate the last example on the link above. I can get validation to work if I use data-toggle="buttons" but then angular binding breaks. 

Do you have any example to perform validation on radio buttons using Bootstrap's button plugin.

**Comment by sagrawal31** [comment]
Let me get back to you on this issue

**Comment by sagrawal31** [comment]
What is currently happening and what kind of behavior you are expecting?

**Comment by mattphung** [comment]
Thank you for the response. I cannot seem to get the label and boarder of the radio buttons to turn red when the user forgot to select a value.  Attached is the html file or the code below. Thank you for your help

```
<!doctype html>
<html ng-app="ui.bootstrap.demo"> 
<head>   
    <script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular.js"></script>
    <script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular-animate.js"></script>
    <script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular-sanitize.js"></script>
    <script src="http://angular-ui.github.io/bootstrap/ui-bootstrap-tpls-2.2.0.js"></script>
    <script
        src="C:\TestGround\Validation\bootstrap-angular-validation-master\dist\bootstrap-angular-validation-all.min.js"></script>
    <link href="http://netdna.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet">
</head>
<body>
    <div ng-controller="ButtonsCtrl">
        <form class="form-horizontal" ng-submit="saveMe()">
                <h4>Radio &amp; Uncheckable Radio</h4>   
            <pre>{{radioModel || 'null'}}</pre>
            <div class="form-group">
                <label for="inputPassword3" class="col-sm-2 control-label">Alignment</label>
                <div class="btn-group">  
                    <label class="btn btn-primary" ng-model="radioModel" uib-btn-radio="'Left'" required="required">
                        Left
                    </label>
                    <label class="btn btn-primary" ng-model="radioModel" uib-btn-radio="'Middle'" required="required">
                        Middle
                    </label>
                    <label class="btn btn-primary" ng-model="radioModel" uib-btn-radio="'Right'" required="required">
                        Right
                    </label>
                </div>
            </div>
               
            <div class="form-group">     
                <div class="col-sm-offset-2 col-sm-10">       
                    <button type="submit" class="btn btn-default">Sign in</button>
                </div>
            </div>
        </form>
    </div>
    <script>
        angular.module('ui.bootstrap.demo', ['ngAnimate', 'ngSanitize', 'ui.bootstrap', 'bootstrap.angular.validation']);
        angular.module('ui.bootstrap.demo').controller('ButtonsCtrl', function ($scope) {
            $scope.singleModel = 1;
            $scope.radioModel = '';
            $scope.checkModel = {left: false, middle: true, right: false};

            $scope.saveMe = function () {
                alert("Form is valid. No errors");
            };

            $scope.checkResults = [];

            $scope.$watchCollection('checkModel', function () {
                $scope.checkResults = [];
                angular.forEach($scope.checkModel, function (value, key) {
                    if (value) {
                        $scope.checkResults.push(key);
                    }
                });
            });
        });
    </script> 
</body>
</html>
```

**Comment by sagrawal31** [comment]
Alright, so there are quite a few problems here. First see a working example below:

http://codepen.io/shashank-agrawal/pen/XjQxgL

Try to directly hit the submit button.
1. The validation auto works only with `input`, `select` and `textarea` element having the `ng-model` applied and since you are using `label` as the input element (which is really not a input element) you need to add `bs-validation` directive on those 3 radio button label
2. The default trigger for displaying validation state is when you blur (remove focus) from a input field but since this is a label, there is no case of blur so I changed the validation trigger to `submit` (check the `.config` block in javascript)
3. By default the error message is appended just after the invalid element but since we have a radio group, you have to add an empty `div` inside the `.form-group` like `<div class="bs-invalid-msg help-block"></div>`. Now the library will use this `div` to display error
4. Since this is a generic library based on Bootstrap, it won't give you additional CSS to style the btn-group so I added a custom style in the CSS section.

(Your markup was wrong for using `form-horizontal` class)

**Comment by sagrawal31** [comment]
Closing this issue as it is not a bug and working as expected. Feel free to comment 😄

**Comment by mattphung** [comment]
Thank you for the quick reply. You're awesome.

```
  From: Shashank Agrawal <notifications@github.com>
```

 To: sagrawal14/bootstrap-angular-validation bootstrap-angular-validation@noreply.github.com 
Cc: mattphung mattphung@yahoo.com; Author author@noreply.github.com
 Sent: Wednesday, October 26, 2016 7:52 PM
 Subject: Re: [sagrawal14/bootstrap-angular-validation] Issue validating Bootstrap (#17)

Alright, so there are quite a few problems here. First see a working example below:http://codepen.io/shashank-agrawal/pen/XjQxgLTry to directly hit the submit button.   
- The validation auto works only with input, select and textarea element having the ng-model applied and since you are using label as the input element (which is really not a input element) you need to add bs-validation directive on those 3 radio button label
- The default trigger for displaying validation state is when you blur (remove focus) from a input field but since this is a label, there is no case of blur so I changed the validation trigger to submit (check the .config block in javascript)
- By default the error message is appended just after the invalid element but since we have a radio group, you have to add an empty div inside the .form-group like <div class="bs-invalid-msg help-block"></div>. Now the library will use this div to display error
- Since this is a generic library based on Bootstrap, it won't give you additional CSS to style the btn-group so I added a custom style in the CSS section.
  (Your markup was wrong for using form-horizontal class)—
